lib/Sema/CSSimplify.cpp

```diff
@@ -3825,8 +3826,41 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     // FIXME(diagnostics): If there were no viable results, but there are
     // unviable ones, we'd have to introduce fix for each specific problem.
-    if (!result.UnviableCandidates.empty())
+    if (!result.UnviableCandidates.empty()) {
+      // Check if we have unviable candidates whose reason for rejection
+      // is either UR_InstanceMemberOnType or UR_TypeMemberOnInstance
+      auto unviable = [&](const std::pair<OverloadChoice,
+                          MemberLookupResult::UnviableReason> &e) {
+        return e.second == MemberLookupResult::UR_InstanceMemberOnType ||
+               e.second == MemberLookupResult::UR_TypeMemberOnInstance;
+      };
+      
+      // If we do, then allow them
+      if (llvm::all_of(result.UnviableCandidates, unviable)) {
+        
+        auto *fix = AllowTypeOrInstanceMember::create(*this, baseTy, member,
+                                                      getConstraintLocator(locator));
+        
+        if (recordFix(fix)) {
+          // We couldn't record a fix, so bail out
+          return SolutionKind::Error;
+        }
+        
+        // The fix was successful, so let's add it to the overload set
+        // and return the solution as solved
+        auto overloadChoices = SmallVector<OverloadChoice, 4>();
+        
+        for (auto &pair : result.UnviableCandidates) {
+          overloadChoices.push_back(pair.first);
+        }
+        
+        addOverloadSet(memberTy, ArrayRef<OverloadChoice>(overloadChoices), useDC,
```

Comment:
I wounder why it's necessary to add `ArrayRef<OverloadChoice>` here? There is an `ArrayRef` constructor which takes `SmallVector` already as far as I remember...

Reply:
I was able to pass `overloadChoices` directly but it led to a crash: `Assertion failed: (decl), function operator(), file /Users/suyashsrijan/Documents/swift-src/swift/lib/Sema/ConstraintSystem.cpp, line 1404.`. 

Creating a `ArrayRef<OverloadChoice>` using it and then passing it fixed the issue - not sure why!

Reply:
Probably Overload decl didn’t get properly copy constructed or something, try moving them into new vector, let’s see if that fixes this issue